### ISSUE TITLE
[WhoReacted] Plugin could not be compiled

### ISSUE THREAD

**Issue by Phanzor** [jaimeadf/BetterDiscordPlugins#361]
```
TypeError: Cannot read property 'default' of undefined
    at eval (betterdiscord://plugins/WhoReacted.plugin.js:166:124)
    at buildPlugin (betterdiscord://plugins/WhoReacted.plugin.js:434:12)
    at eval (betterdiscord://plugins/WhoReacted.plugin.js:441:7)
    at Object.requireAddon (<anonymous>:4:43095)
    at Object.loadAddon (<anonymous>:4:8859)
    at Object.loadAddon (<anonymous>:4:40847)
    at Object.updateList (<anonymous>:4:10713)
    at Object.updatePluginList (<anonymous>:4:40544)
    at O.reload (<anonymous>:4:108624)
    at Object.Ye (https://discord.com/assets/417057e696fbefab8369.js:6149:658)
    at qe (https://discord.com/assets/417057e696fbefab8369.js:6149:812)
    at https://discord.com/assets/417057e696fbefab8369.js:6169:409
    at Pr (https://discord.com/assets/417057e696fbefab8369.js:6169:506)
    at Mr (https://discord.com/assets/417057e696fbefab8369.js:6169:920)
    at https://discord.com/assets/417057e696fbefab8369.js:6175:680
    at ke (https://discord.com/assets/417057e696fbefab8369.js:6257:404)
    at https://discord.com/assets/417057e696fbefab8369.js:6171:395
    at Br (https://discord.com/assets/417057e696fbefab8369.js:6171:425)
    at rn (https://discord.com/assets/417057e696fbefab8369.js:6158:757)
    at nn (https://discord.com/assets/417057e696fbefab8369.js:6158:7)
    at t.unstable_runWithPriority (https://discord.com/assets/417057e696fbefab8369.js:6369:828)
    at Ki (https://discord.com/assets/417057e696fbefab8369.js:6180:117)
    at Ue (https://discord.com/assets/417057e696fbefab8369.js:6257:144)
    at en (https://discord.com/assets/417057e696fbefab8369.js:6157:800)
    at HTMLDivElement.r (https://discord.com/assets/ea40f89e67e976efa166.js:126:559)
```

**Comment by leranjun** [comment]
Can reproduce

**Comment by zhurkin** [comment]
same problem

TypeError: Cannot read property 'default' of undefined
    at eval (betterdiscord://plugins/WhoReacted.plugin.js:166:124)
    at buildPlugin (betterdiscord://plugins/WhoReacted.plugin.js:434:12)
    at eval (betterdiscord://plugins/WhoReacted.plugin.js:441:7)
    at Object.requireAddon (<anonymous>:4:43095)
    at Object.loadAddon (<anonymous>:4:8859)
    at Object.loadAddon (<anonymous>:4:40847)
    at Object.loadAllAddons (<anonymous>:4:11269)
    at Object.initialize (<anonymous>:4:5963)
    at Object.initialize (<anonymous>:4:40050)
    at Object.startup (<anonymous>:4:374950)

**Comment by medovanx** [comment]
same issue
`TypeError: Cannot read property 'default' of undefined
    at eval (betterdiscord://plugins/WhoReacted.plugin.js:166:124)
    at buildPlugin (betterdiscord://plugins/WhoReacted.plugin.js:434:12)
    at eval (betterdiscord://plugins/WhoReacted.plugin.js:441:7)
    at Object.requireAddon (<anonymous>:4:43095)
    at Object.loadAddon (<anonymous>:4:8859)
    at Object.loadAddon (<anonymous>:4:40847)
    at FSWatcher.<anonymous> (<anonymous>:4:7059)`

**Comment by ja-som** [comment]
Same as people above.
```
TypeError: Cannot read property 'default' of undefined
    at eval (betterdiscord://plugins/WhoReacted.plugin.js:166:124)
    at buildPlugin (betterdiscord://plugins/WhoReacted.plugin.js:434:12)
    at eval (betterdiscord://plugins/WhoReacted.plugin.js:441:7)
    at Object.requireAddon (<anonymous>:4:43095)
    at Object.loadAddon (<anonymous>:4:8859)
    at Object.loadAddon (<anonymous>:4:40847)
    at Object.loadAllAddons (<anonymous>:4:11269)
    at Object.initialize (<anonymous>:4:5963)
    at Object.initialize (<anonymous>:4:40050)
    at Object.startup (<anonymous>:4:374950)
```

**Comment by NoodleSushi** [comment]
Same issue
```
TypeError: Cannot read property 'default' of undefined
    at eval (betterdiscord://plugins/WhoReacted.plugin.js:166:124)
    at buildPlugin (betterdiscord://plugins/WhoReacted.plugin.js:434:12)
    at eval (betterdiscord://plugins/WhoReacted.plugin.js:441:7)
    at Object.requireAddon (<anonymous>:4:43095)
    at Object.loadAddon (<anonymous>:4:8859)
    at Object.loadAddon (<anonymous>:4:40847)
    at Object.loadAllAddons (<anonymous>:4:11269)
    at Object.initialize (<anonymous>:4:5963)
    at Object.initialize (<anonymous>:4:40050)
    at Object.startup (<anonymous>:4:374950)
```

**Comment by SpiffyJUNIOR** [comment]
Same here

**Comment by M0REKZ** [comment]
i think i dont need to say same here too

**Comment by XordK** [comment]
Today is not a good day for the contributors lol. Same issue here.

**Comment by Th3FiN4LSt4rT** [comment]
Same Issue, at each time I open Discord ( On Boot or reboot of my computer) i got this too
![image](https://user-images.githubusercontent.com/113826247/190893803-e3febadc-5c06-4283-bcbe-6e4e44ba55ae.png)

**Comment by 9EED** [comment]
same issue

**Comment by ashap5** [comment]
same issue xd

**Comment by x04ty29er** [comment]
It looks like the "Reactions" module being looked for at line 166 got renamed to "MessageReactions".
As a temporary workaround, you should be able to manually patch this line and reload.

```
const Reactions = external_BoundedLibrary_namespaceObject.WebpackModules.find(m => m?.default?.displayName === 'Reactions').default;
```
To
```
const Reactions = external_BoundedLibrary_namespaceObject.WebpackModules.find(m => m?.default?.displayName === 'MessageReactions').default;
```

I'm unsure what other effects/changes this may cause. No guarantees.

**Comment by SpiffyJUNIOR** [comment]
> `const Reactions = external_BoundedLibrary_namespaceObject.WebpackModules.find(m => m?.default?.displayName === 'MessageReactions').default;`
Tested, seems to work. Maybe someone should make a PR with the change?

**Comment by merri1400** [comment]
So I feel dumb because the plugin won't compile at the start and it doesn't let me get to that line to change it, it's only a few lines. How do I get that far down in the module?

**Comment by terr0r-exe** [comment]
> It looks like the "Reactions" module being looked for at line 166 got renamed to "MessageReactions". As a temporary workaround, you should be able to manually patch this line and reload.
> 
> ```
> const Reactions = external_BoundedLibrary_namespaceObject.WebpackModules.find(m => m?.default?.displayName === 'Reactions').default;
> ```
> 
> To
> 
> ```
> const Reactions = external_BoundedLibrary_namespaceObject.WebpackModules.find(m => m?.default?.displayName === 'MessageReactions').default;
> ```
> 
> I'm unsure what other effects/changes this may cause. No guarantees.

didn't work for me 😭

**Comment by x04ty29er** [comment]
> So I feel dumb because the plugin won't compile at the start and it doesn't let me get to that line to change it, it's only a few lines. How do I get that far down in the module?

For those that need some assistance patching while we wait for a fix:
1. Open Discord. Plugin will fail to load and you will get the error message. This is ok.
2. Open Settings > BetterDiscord > Plugins
3. Click "Open Plugins Folder" button
4. Find WhoReacted.plugin.js and right click > Open With > Notepad
5. Apply the fix mentioned above and save the file. This should trigger a reload. Otherwise, CTRL+R should force a reload.
6. Plugin should load correctly. If it doesn't, you will hopefully see a difference exception that indicates a different issue.

👍

**Comment by merri1400** [comment]
It still only lets me get to line 6 on Notepad, I can't get to line 133.

**Comment by merri1400** [comment]
This is all that's in the file:
{
    "currentVersionInfo": {
        "version": "1.2.5",
        "hasShownChangelog": true
    }
}

**Comment by merri1400** [comment]
Found it, thanks. I don't usually dig around in the files themselves, just install them and drag.

**Comment by Visne** [comment]
Fixed in 63d8e47354e4309b4d0408ea210b488fd4f8434d
